src/main/java/org/broadinstitute/consent/http/mail/message/MailMessage.java

```diff
@@ -1,19 +1,29 @@
 package org.broadinstitute.consent.http.mail.message;
 
-import com.sendgrid.helpers.mail.Mail;
-import com.sendgrid.helpers.mail.objects.Content;
-import com.sendgrid.helpers.mail.objects.Email;
-import java.io.Writer;
+import org.broadinstitute.consent.http.enumeration.EmailType;
+import org.broadinstitute.consent.http.models.User;
 
 public abstract class MailMessage {
 
-  protected Mail generateEmailMessage(String toAddress, String fromAddress, Writer template,
-      String referenceId, String type) {
-    Content content = new Content("text/html", template.toString());
-    String subject = assignSubject(referenceId, type);
-    return new Mail(new Email(fromAddress), subject, new Email(toAddress), content);
+  public final User toUser;
+  public final EmailType emailType;
+
+  protected MailMessage(User toUser, EmailType emailType) {
+    this.toUser = toUser;
+    this.emailType = emailType;
+  }
+
+  public String getTemplateName() {
+    return emailType.templateName;
   }
 
-  abstract String assignSubject(String referenceId, String type);
+  public abstract String createSubject();
 
-}
\ No newline at end of file
+  public abstract Object createModel(String serverUrl);
```

Comment:
Having this accept `serverUrl` is awkward; I think this could be fixed by having this return `Map<String, Object>` and the caller can add `serverUrl` to the Map.